service/ciphering/encoder.go

```diff
@@ -0,0 +1,6 @@
+package ciphering
+
+type Encoder interface {
+	Encrypt(plaintext string, aesKey []byte, useRandom bool) (string, error)
+	Decrypt(ct string, aesKey []byte, useRandom bool) ([]byte, error)
```

Comment:
Why do we need it? We haven't discussed that, so lean only to agreed changes/functionality.

```suggestion
	Decrypt(ct string, aesKey []byte) ([]byte, error)
```

Reply:
while adding  rewrite by password functionality i was not able to encode password into same value twice or more time because of randomized part of encoding function, which provides higher level of security. Thats the reason why i have splited encoding flow into to ways: one with randomized part(for value encoding), another one without this part(for key encoding). To understand which way should certain data be encoded i added bool "flags" (useRandom bool)

Reply:
> was not able to encode password into same value twice or more time because of randomized part of encoding function, which provides higher level of security.

So how had then this function works before? It is the same algorithm but instead of using structure wide variable it uses function wide variable? Remove please this flag and keep to our agreements.

Reply:
> 

all changes related to this problem are in the [last commit](https://github.com/retttddd/golangProjectv3/pull/1/commits/44d0c73d36187fa2398318dd7e5b9cc1184551a9)